pkg/providers/v1/aws_test.go

```diff
@@ -4017,3 +4040,232 @@ func TestIsAWSErrorInstanceNotFound(t *testing.T) {
+							Timeout:            aws.Int64(5),
+							UnhealthyThreshold: aws.Int64(2),
+							HealthyThreshold:   aws.Int64(2),
+						},
+					},
+				},
+			}, nil)
+			awsServices.elb.(*MockedFakeELB).On("ConfigureHealthCheck", &elb.ConfigureHealthCheckInput{
+				LoadBalancerName: aws.String("aid"),
+				HealthCheck: &elb.HealthCheck{
+					Target:             aws.String("TCP:31173"),
+					Interval:           aws.Int64(10),
+					Timeout:            aws.Int64(5),
+					UnhealthyThreshold: aws.Int64(6),
+					HealthyThreshold:   aws.Int64(2),
+				},
+			}).Return(&elb.ConfigureHealthCheckOutput{}, nil)
+
+			tags := []ec2types.Tag{
+				{Key: aws.String(TagNameKubernetesClusterLegacy), Value: aws.String(TestClusterID)},
+				{Key: aws.String(fmt.Sprintf("%s%s", TagNameKubernetesClusterPrefix, TestClusterID)), Value: aws.String(ResourceLifecycleOwned)},
+			}
+			awsServices.ec2.(*MockedFakeEC2).On("DescribeInstances", &ec2.DescribeInstancesInput{
+				InstanceIds: []string{"i-2bce90670bb0c7ce", "i-2bce90670bb0c7cf"},
+			}).Return([]ec2types.Instance{
+				{
+					InstanceId: aws.String("i-2bce90670bb0c7ce"),
+					SecurityGroups: []ec2types.GroupIdentifier{
+						{GroupId: aws.String(fakeSecurityGroupID)},
+					},
+					Tags: tags,
+				},
+				{
+					InstanceId: aws.String("i-2bce90670bb0c7cf"),
+					SecurityGroups: []ec2types.GroupIdentifier{
+						{GroupId: aws.String(fakeSecurityGroupID)},
+					},
+					Tags: tags,
+				},
+			}, nil)
+
+			awsServices.ec2.(*MockedFakeEC2).On("DescribeSecurityGroups", &ec2.DescribeSecurityGroupsInput{}).Maybe().Return(
+				[]ec2types.SecurityGroup{{GroupId: aws.String(fakeSecurityGroupID), Tags: tags}},
+			)
+
+			// Configure the tests
+			cfg := config.CloudConfig{}
+			if test.config != nil {
+				cfg = test.config()
+			}
+			c, err := newAWSCloud(cfg, awsServices)
+			assert.Nil(t, err, "Error building aws cloud: %v", err)
+
+			awsServices.ec2.(*MockedFakeEC2).Subnets = []ec2types.Subnet{
+				{
+					AvailabilityZone: aws.String("us-west-2a"),
+					SubnetId:         aws.String("subnet-abc123de"),
+					Tags: []ec2types.Tag{
+						{
+							Key:   aws.String(c.tagging.clusterTagKey()),
+							Value: aws.String("owned"),
+						},
+					},
+				},
+			}
+			awsServices.ec2.(*MockedFakeEC2).RouteTables = []ec2types.RouteTable{
+				{
+					Associations: []ec2types.RouteTableAssociation{
+						{
+							Main:                    aws.Bool(true),
+							RouteTableAssociationId: aws.String("rtbassoc-abc123def456abc78"),
+							RouteTableId:            aws.String("rtb-abc123def456abc78"),
+							SubnetId:                aws.String("subnet-abc123de"),
+						},
+					},
+					RouteTableId: aws.String("rtb-abc123def456abc78"),
+					Routes: []ec2types.Route{
+						{
+							DestinationCidrBlock: aws.String("0.0.0.0/0"),
+							GatewayId:            aws.String("igw-abc123def456abc78"),
+							State:                ec2types.RouteStateActive,
+						},
+					},
+				},
+			}
```

Comment:
Check if can move up those two last calls to before the line [4208](https://github.com/kubernetes/cloud-provider-aws/pull/1158/files#diff-5c51f8c2333051ed351b49b7bb3242551e72ec906d3fe77866e71509a1880b93R4208) to make it more readable.